armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -488,96 +508,130 @@ def _getLinkedComponents(self, b, c):
+            component to compare and see if is linked to componentA
 
-    Parameters
-    ----------
-    componentA : :py:class:`Component <armi.reactor.components.component.Component>`
-        component of interest
-    componentB : :py:class:`Component <armi.reactor.components.component.Component>`
-        component to compare and see if is linked to componentA
+        Notes
+        -----
+        If componentA and componentB are both solids and the same type, geometric overlap can be checked via
+        getCircleInnerDiameter and getBoundingCircleOuterDiameter. Five different cases are accounted for.
+        If they do not meet these initial criteria, linkage is assumed to be False.
+        Case #1: Unshaped Components. There is no way to determine overlap so they're assumed to be not linked.
+        Case #2: Blocks with specified grids. If componentA and componentB share common grid indices (cannot be a partial
+        case, ALL of the indices must be contained by one or the other), then overlap can be checked.
+        Case #3: If Component position is not specified via a grid, the multiplicity is checked. If consistent, they are
+        assumed to be in the same positions and their overlap is checked.
+        Case #4: Cases 1-3 are not True so we assume there is no linkage.
+        Case #5: Components are either not both solids or are not the same type. These cannot be linked.
+
+        Returns
+        -------
+        linked : bool
+            status is componentA and componentB are axially linked to one another
+        """
+        if isinstance(componentA, type(componentB)):
+            if isinstance(componentA, UnshapedComponent):
+                ## Case 1 -- see docstring
+                runLog.warning(
+                    f"Components {componentA} and {componentB} are UnshapedComponents "
+                    "and do not have 'getCircleInnerDiameter' or getBoundingCircleOuterDiameter methods; "
+                    "nor is it physical to do so. Instead of crashing and raising an error, "
+                    "they are going to be assumed to not be linked.",
+                    single=True,
+                )
+                linked = False
+            elif isinstance(
+                componentA.spatialLocator, MultiIndexLocation
+            ) and isinstance(componentB.spatialLocator, MultiIndexLocation):
+                ## Case 2 -- see docstring
+                componentAIndices = [
+                    list(index) for index in componentA.spatialLocator.indices
+                ]
+                componentBIndices = [
+                    list(index) for index in componentB.spatialLocator.indices
+                ]
+                # check for common indices between components. If either component has indices within its counterpart,
+                # then they are candidates to be linked and overlap should be checked.
```

Comment:
Just thinking out loud here... It seems like this logic will not allow for partial-length fuel rods, if I'm reading it correctly. That's not an issue for me right now, but just trying to consider common cases (partial-length rods are common in BWRs).